### PR TITLE
Exposing internals of SqlClient for Unittesting

### PR DESCRIPTION
**Pull request by saurabh500**
The PR exposes the internals of SqlClient to a Unit Tests project.

I considered exposing the internals to FunctionalTests, but there were Type ambiguities found.
This would have required a significant refactoring of the client.

As a result, I went ahead with creating a new project. 

Next steps: Get this project to run as part of the pipeline.

### CODE REVIEW

**Review by benrr101 — APPROVED** [review]
If you can get the builds working, feel free to merge

File: src/Microsoft.Data.SqlClient/netcore/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿using System;
+using System.Runtime.CompilerServices;
```
Comment (VladimirReshetnikov) [inline src/Microsoft.Data.SqlClient/netcore/src/Properties/AssemblyInfo.cs:2]: `using System.Runtime.CompilerServices;` is the only directive that is actually needed, the others seem redundant.

**Comment by cheenamalhotra** [comment]
Checking in, is this still needed?

**Comment by saurabh500** [comment]
Maybe not

**Pull request closed without merging**